src/course/infrastructure/repository/course-mock-repository.ts

```diff
@@ -0,0 +1,89 @@
+import { CourseRepository } from '../../domain/repository/course-repository';
+import { Course } from '../../domain/course';
+import { CourseId } from '../../domain/course-id';
+import { CourseFilters } from 'course/domain/course-filters';
+
+export class CourseMockRepository implements CourseRepository {
+  private courses: Course[] = [];
+
+  create(course: Course): Promise<Course> {
+    this.courses.push(course);
+    return Promise.resolve(course);
+  }
+
+  async getAll(
+    page: number = 1,
+    limit: number = 10,
+    filters?: CourseFilters
+  ): Promise<{ courses: Course[]; total: number }> {
+    let filteredCourses = [...this.courses];
+
+    if (filters?.level) {
+      filteredCourses = filteredCourses.filter(
+        course => course.level === filters.level
+      );
+    }
+    if (filters?.instructor) {
+      filteredCourses = filteredCourses.filter(
+        course => course.instructor === filters.instructor
+      );
+    }
+
+    const total = filteredCourses.length;
+
+    const startIndex = (page - 1) * limit;
+    const endIndex = startIndex + limit;
+    const paginatedCourses = filteredCourses.slice(startIndex, endIndex);
+
+    return Promise.resolve({
+      courses: paginatedCourses,
+      total,
+    });
+  }
+
+  getById(id: CourseId): Promise<Course | null> {
+    const course = this.courses.find(
+      course => course.id.getValue() === id.getValue()
+    );
+    return Promise.resolve(course || null);
+  }
+
+  delete(id: CourseId): Promise<void> {
+    this.courses = this.courses.filter(
+      course => course.id.getValue() !== id.getValue()
+    );
+    return Promise.resolve();
+  }
+
+  edit(id: CourseId, courseData: Partial<Course>): Promise<Course> {
+    const course = this.courses.find(
+      course => course.id.getValue() === id.getValue()
+    );
+
+    if (!course) {
+      throw new Error('Course not found');
+    }
+
+    if (courseData.title !== undefined) {
+      course.setTitle(courseData.title);
```

Comment:
course.setTitle(courseData.title) passes a raw string to setTitle, but Course.setTitle expects a CourseTitle (a domain value object). This leaks domain invariants and will cause runtime errors or bypass validation.

**Sugerencia:**
```typescript
Either accept plain strings in setTitle / setDescription (and perform validation inside Course) or create the VO here before setting it. For example:
import { CourseTitle } from '../../domain/course-title';
...
if (courseData.title !== undefined) {
  course.setTitle(new CourseTitle(courseData.title));
}
Alternatively, change setTitle signature to accept a string and construct the CourseTitle inside Course.
```